src/classes/vehicle.py

```diff
@@ -87,73 +90,123 @@ def update_cmd(old_cmd: Command, t: np.array, a: np.array, elapsed_time: float=0
 
 def driver_traffic_update_command(vehicles: list, cur_time: float) -> None:
     """Update command for standard traffic."""
-
-    # calculate command that achieves two things
-    # 1. stop before any traffic stop line if that traffic light is red
-    #    traffic light red? stop before it
-    #    traffic light green? speed up
-    # 2. maintain distance between car with closest collision (the car in front)
-    #    if conditions match to calculate new command
-    #    a. is our distance to that car within X?
-    #       is their acceleration lower than ours?
-    #       -> calculate slow down
-    #    b. is our distance to that car more than Y?
-    #       is their acceleration more than ours?
-    #       -> calculate speed up
-
-    # 1 and 2 need to work together. If 2. determines that we can speed up,
-    # but the car is approaching a red traffic light,
-    # then the final command should be to slow down
-    
     update_driver_lead(vehicles)
+
     for vehicle in vehicles:
-        leading_vehicle = vehicle.leading_vehicle
-        initial_velocity = vehicle.velocity
-        
-        if leading_vehicle:
-            final_velocity = leading_vehicle.velocity
-            distance = abs(vehicle.route_position - leading_vehicle.route_position) 
+        if vehicle.leading_vehicle:
+            drive_vehicle_with_leading(vehicle, cur_time)
+        # only accelerate when the vehicle is on a StraightEdge
+        elif vehicle.direction_angle % 90 == 0:
```

Comment:
The idea is good, it makes sense. But I think vehicles accelerate even while they're still slightly turning. maybe we can have a window of an angle where they would accelerate. +/-10 deg ?
I will let you figure out how to do this :)